adapter/src/ethereum.rs

```diff
@@ -269,6 +271,64 @@ impl Adapter for EthereumAdapter {
         ewt_sign(&wallet, &self.keystore_pwd, &payload)
             .map_err(|err| AdapterError::Adapter(Error::SignMessage(err).into()))
     }
+
+
+    async fn get_spendable(&self, channel: &Channel, spender: &ValidatorId) -> AdapterResult<SpendableOutput, Self::AdapterError> {
+        let outpace_contract = Contract::from_json(
+            self.web3.eth(),
+            self.config.ethereum_core_address.into(),
+            &OUTPACE_ABI,
+        )
+        .map_err(Error::ContractInitialization)?;
+
+        let erc20_contract = Contract::from_json(
+            self.web3.eth(),
+            self.config.ethereum_core_address.into(),
+            &ERC20_ABI,
+        )
+        .map_err(Error::ContractInitialization)?;
+
+        let mut amount: U256 = tokio_compat_02::FutureExt::compat(async {
+            tokio_compat_02::FutureExt::compat(outpace_contract.query(
+                "deposits",
+                (channel.id.into_token(), spender.to_hex_prefix_string().into_token()),
```

Comment:
You need these types to be from the following list:

https://docs.rs/web3/0.15.0/web3/contract/tokens/trait.Tokenize.html?search=into_token

- The `ChannelId` has `[u8; 32]` (`FixedBytes` or known as `bytes32` in the contact )
- The `ValidatorId` hsa `[u8; 20]` (or `Address`)
You can also see those types in the contract:

https://github.com/AdExNetwork/adex-protocol-eth/blob/identity-tests/contracts/OUTPACE.sol#L43
It's possible that the this will fail if you try to test it.